docs/ActiveLabel.rst

```diff
@@ -0,0 +1,135 @@
+ActiveLabel
+===========
+
+``ActiveNode`` provides an ability to define inheritance of models which also gives subclasess the labels of their parent models.  In Ruby, however, inheritence of classes is not sufficient.  Sometimes is makes more sense to be able to build a module which defines behavior (or "concerns") which could be applied to any model.  This is what ``ActiveLabel`` provides.
+
+``ActiveLabel`` modules can be defined in two ways:
+
+ * Default: Where the module's behavior is always defined on the ``ActiveNode`` model and the model's nodes always have a corresponding label in Neo4j
+ * Optional: Where the module's behavior is defined on the class only when the model's nodes have a corresponding label in Neo4j
+
+.. code-block:: ruby
+
+  class Person
+    include Neo4j::ActiveNode
+
+    property :name, type: String
+
+    label :HasAddress
+    label :Destroyable
+  end
+
+  class Organization
+    include Neo4j::ActiveNode
+
+    property :title, type: String
+
+    label :HasAddress
+    label :Destroyable
+  end
+
+
+.. code-block:: ruby
+
+  class Address
+    property :line1, type: String
+    property :line2, type: String
+    property :country, type: String
+    property :postal_code, type: String
+  end
+
+  module HasAddress
+    include Neo4j::ActiveLabel
+
+    included do
+      has_one :out, :address, type: :HAS_ADDRESS
+    end
+
+    module InstanceMethods
+      def distance_from(has_address_object)
+        address.distance_from(has_address_object.address)
+      end
+    end
+  end
+
+
+.. code-block:: ruby
+
+  module Destroyable
+    include Neo4j::ActiveLabel
+
+    follows_label :Destroyed
```

Comment:
@leehericks Responding to your comment about following the module name

I'm definitely a bit on the fence about having a `Destroyable` module which defines a `Destroyed` label, but it was a choice that I made on purpose because I'm trying to keep a Ruby style.  In Ruby you often have module names which are adjectives like `Destroyable` /`Touchable` / `Whateverable`, but that doesn't make sense as a label name.

The question of what `Destroyable.all` returns definitely makes me take a step back.  Maybe it's the case that all of these nodes get a `Destroyable` label but then you can define something like `boolean_label :Destroyed` instead of `follows_label :Destroyed` would let you optionally add the `Destroyed` label to nodes and would also allow you to query for `Destroyed` nodes.

As I'm thinking about this more and more, this might actually be best implemented as two features.  `ActiveLabel` might just be something which always adds the label to the module that it's defined on and we have a method like `boolean_label` which you can apply to `ActiveNode` modules (perhaps even via the `included` block of a module) which allows you to define labels which can be added, removed, and queried/filtered upon.

That could actually simplify defining `ActiveLabel`s as well because rather than doing `label :Destroyable` in the model we could simply do `include Destroyable` and use Ruby's normal module logic to bring in the behavior.  I was using the `label` method so that methods could be defined optionally.

So maybe then it does make sense to think in terms similar to what @thefliik proposed to have something like:

```ruby
optional_label :Destroyed do
  property :destroyed_at, type: DateTime

  def destroy
    destroyed_at = Time.now

    super
  end
end
```

That could be defined on the model or inside of a `include`d module

Reply:
Oops, I should have read this first.
> That could actually simplify defining ActiveLabels as well because rather than doing label :Destroyable in the model we could simply do include Destroyable and use Ruby's normal module logic to bring in the behavior.

Looks like we've arrived at similar conclusions